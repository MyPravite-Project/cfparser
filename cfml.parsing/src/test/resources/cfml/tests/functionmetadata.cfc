function setStatus( rc ) my:role="support:admin" {}
function setStatus2( rc ) role="support:admin" {}
function setStatus2( rc ) role {}
function setStatus2( rc ) role:support {}