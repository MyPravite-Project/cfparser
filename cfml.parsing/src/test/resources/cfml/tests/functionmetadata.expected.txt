/*===TOKENS===*/
FUNCTION                 <function>
IDENTIFIER               <setStatus>
'('                      <(>
IDENTIFIER               <rc>
')'                      <)>
IDENTIFIER               <my>
':'                      <:>
IDENTIFIER               <role>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <support:admin>
CLOSE_STRING             <">
'{'                      <{>
'}'                      <}>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <setStatus2>
'('                      <(>
IDENTIFIER               <rc>
')'                      <)>
IDENTIFIER               <role>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <support:admin>
CLOSE_STRING             <">
'{'                      <{>
'}'                      <}>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <setStatus2>
'('                      <(>
IDENTIFIER               <rc>
')'                      <)>
IDENTIFIER               <role>
'{'                      <{>
'}'                      <}>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <setStatus2>
'('                      <(>
IDENTIFIER               <rc>
')'                      <)>
IDENTIFIER               <role>
':'                      <:>
IDENTIFIER               <support>
'{'                      <{>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (element
    (functionDeclaration
      function
      (identifier setStatus)
      (
      (parameterList (parameter (identifier rc)))
      )
      (functionAttribute
        (identifierWithColon (identifier my) : (identifier role))
        =
        (constantExpression (stringLiteral " (stringLiteralPart support:admin) "))
      )
      (compoundStatement { })
    )
  )
  (element
    (functionDeclaration
      function
      (identifier setStatus2)
      (
      (parameterList (parameter (identifier rc)))
      )
      (functionAttribute
        (identifier role)
        =
        (constantExpression (stringLiteral " (stringLiteralPart support:admin) "))
      )
      (compoundStatement { })
    )
  )
  (element
    (functionDeclaration
      function
      (identifier setStatus2)
      (
      (parameterList (parameter (identifier rc)))
      )
      (functionAttribute (identifier role))
      (compoundStatement { })
    )
  )
  (element
    (functionDeclaration
      function
      (identifier setStatus2)
      (
      (parameterList (parameter (identifier rc)))
      )
      (functionAttribute (identifier role) : (identifier support))
      (compoundStatement { })
    )
  )
)
/*======*/
/*===DECOMPILE===*/
{
  public function setStatus2(rc) role='support:admin'    {

    };
  public function setStatus(rc) my:role='support:admin'    {

    };
  public function setStatus2(rc) role    {

    };
  public function setStatus2(rc) role=support    {

    };

}
/*======*/